web/src/components/MapView.tsx

```diff
@@ -68,18 +69,16 @@ const MapView: React.FunctionComponent<MapViewProps> = (props: MapViewProps): Re
           latitude: geometry.coordinates[1]!,
           zoom: detailZoom
         }))
-        setCurrentFeature(currentFeature)
-        togglePopup(true)
+        selectFeature(currentFeature)
       }
     }
-  }, [featureCollection, queryLocation])
+  }, [featureCollection, queryLocation, selectFeature])
 
   const clickItem = (e: MapEvent) => {
     if (e.features?.length) {
-      setCurrentFeature(e.features[0])
-      togglePopup(true)
+      selectFeature(e.features[0], 1)
```

Comment:
Ternary would be nicer `
  const clickItem = (e: MapEvent) => (e.features?.length ? selectFeature(e.features[0], 1) : selectFeature(null))`

Reply:
i splitted up some functions like `changeSnapPoint` and had to `updateQueryParams` so it can not be done in tenary expression. sorry for the meantime update. i recognized some updated comments after approval from @klinzo